test/e2e/network/service.go

```diff
@@ -4054,13 +4060,18 @@ var _ = SIGDescribe("SCTP [Feature:SCTP] [LinuxOnly]", func() {
 
 		err = e2enetwork.WaitForService(f.ClientSet, ns, serviceName, true, 5*time.Second, e2eservice.TestTimeout)
 		framework.ExpectNoError(err, fmt.Sprintf("error while waiting for service:%s err: %v", serviceName, err))
-
+		// wait until kube-proxy syncs rules
+		// TODO: make this more reliable
+		time.Sleep(5 * time.Second)
```

Comment:
`e2enetwork.WaitForService` waits for the Service to be created, but we should wait for the _Endpoints_ to be created (which in turn implies that both the Service and the Pod have been created). There's a method for that. After that, kube-proxy is allowed to take up to `e2eservice.KubeProxyEndpointLagTimeout` to update iptables, but you should poll rather than just sleeping that whole time. There should be examples around somewhere.

Reply:
I thought the same, but IIUIC this test does not create endpoints, it just checks for the service reject rule.
I've checked locally with udp service without pods associated
```
-A KUBE-SERVICES -d 10.96.57.5/32 -p udp -m comment --comment "default/my-service: has no endpoints" -m udp --dport 80 -j REJECT --reject-with icmp-port-unreachable
```

Reply:
ok, in that case it should be `e2eservice.KubeProxyLagTimeout` (which is weirdly high and should probably be fixed, but anyway, definitely poll rather than sleep)

Reply:
polling is the right way, I was a bit lazy here, sorry